airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -245,3 +247,40 @@ def set_autocommit(self, conn, autocommit: Any) -> None:
 
     def get_autocommit(self, conn):
         return getattr(conn, 'autocommit_mode', False)
+
```

Comment:
Related to point 1 - this problem (allow multiple queries) is not unique to Snowflake. This also applies to Presto and Trino so whatever solution that will be accepted here will most likely be copied to the other two providers as well.

2. I'm not sure if

Reply:
Thanks for the feedback, I agree with both points and will adjust. 
I did not find better way than copying without changing DBApi semantics, also, when looking for similar solution saw that that's what some connectors do already - like Exasol one.

Reply:
We could potentially add some "callbacks" in the DBApi implementation, but this would mean Airflow 2.1+ compatibility. I am not sure it's worth it. Maybe we could figure out some ways of doing back-compatibility and implement it "properly" (like copying the method but use calbacks/original run with callbacks() when available? But I am not sure if it is worth it.

Reply:
That could possibly allow native [OpenLineage](https://github.com/OpenLineage/OpenLineage) integration in airflow, but that's separate topic :slightly_smiling_face: 

>Maybe we could figure out some ways of doing back-compatibility and implement it "properly" (like copying the method but use calbacks/original run with callbacks() when available?

Yes. that makes sense - there's a lot of metadata that some databases expose in a non standard way. One example is `BigQueryExecuteQueryOperator`, which puts job id into xcoms. 
For now, I don't think exposing query id in this particular way in this particular provider will make it in the future.

Reply:
Open Lineage integration is a cool idea. @bolkedebruin might be interested :) . How about starting discussion about it at the devlist @mobuchowski ? Sounds like a very good idea.

Reply:
We're discussing it internally and will come back with some ideas surely :slightly_smiling_face:

Reply:
Is it possible to save the query_ids as the inlets/outlets defined for the LineageBackend? Anyone have an idea of how sending this lineage data to the LineageBackend would be accomplished?